packages/@atjson/editor/src/editor-demo/index.ts

```diff
@@ -0,0 +1,75 @@
+import Document from '@atjson/document';
+import Editor from '../index';
+import WebComponentRenderer from '../webcomponent-renderer';
+import CommonmarkRenderer from '@atjson/renderer-commonmark';
+import InspectorGadget from '../inspector-gadget';
+import OffsetLogo from '../components/logo';
+import events from '../mixins/events';
+
+import EditableLink from '../components/editable-link';
+
+if (!window.customElements.get('text-editor')) {
+  window.customElements.define('text-editor', Editor);
+}
+
+if (!window.customElements.get('inspector-gadget')) {
+  window.customElements.define('inspector-gadget', InspectorGadget);
+}
+
+if (!window.customElements.get('offset-logo')) {
+  window.customElements.define('offset-logo', OffsetLogo);
+}
+
+export default class EditorDemo extends events(HTMLElement) {
+  static template = '<h1><offset-logo offset="0"></offset-logo></h1><text-editor></text-editor>' +
+                    '<h1>HTML Output</h1><div class="output"></div>' + 
+                    '<h1>Commonmark Output</h1><div class="markdown"></div>' +
+                    '<inspector-gadget></inspector-gadget>';
+
+  static events = {
+    'change text-editor'(evt) {
+      this.renderOutput(evt.detail.document);
+      this.renderMarkdown(evt.detail.document);
+    }
+  }
+
+  renderOutput(doc) {
+    let outputElement = this.querySelector('.output');
+    let rendered = new WebComponentRenderer(doc).render();
+    outputElement.innerHTML = rendered.innerHTML;
+  }
+
+  renderMarkdown(doc) {
+    let outputElement = this.querySelector('.markdown');
+    let rendered = new CommonmarkRenderer().render(doc);
+    outputElement.innerHTML = rendered;
+  }
+
+  setDocument(doc: Document) {
+    let editor = this.querySelector('text-editor');
+
+    editor.setDocument(doc);
+
+    let inspectorGadget = this.querySelector('inspector-gadget');
```

Comment:
Hmm - gave this some thought, and I'm not sure. Reasoning: Ideally, the inspector could live in the Chrome dev inspector, and in that case you'd need to be able to set the document separately. It might also be the case that we'd like to be able to open up an inspector after startup, and the machinations to do that with the wrapping version (although elegant) might be overkill?